src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,369 @@ title: Builder.io & Astro
+
+* Make sure the site is live - for example, your dev server is running.
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+* Make sure it's the full URL including the protocol, for example `https://`.
+* If you're working in a virtual environment like [StackBlitz](https://stackblitz.com/) or [Gitpod](https://www.gitpod.io/), you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read [Builder's troubleshooting guide](https://www.builder.io/c/docs/guides/preview-url-working).
+:::
+
+### Creating a blog post
+
+1. In Builder's visual editor, create a new content entry with the following values:
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+2. Complete your post using the **Add Block** button and add a text field with some post content.
+3. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+4. When you're ready click the **Publish** button in the upper right corner.
+5. Create as many posts as you like, ensuring that all content entries contain a `title` and a `slug` as well as some post content.
+
+### Displaying a list of blog posts
+
+Add the following content to `src/pages/index.astro` in order to fetch and display a list of all post titles, each linking to its own page:
+
+```astro title="src/pages/index.astro" {9}
+---
+
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+
+<html lang="en">
+  <head>
+    <title>Blog Index</title>
+  </head>
+  <body>
+    <ul>
+      {
+        posts.map(({ data: { slug, title } }) => (
+          <li>
+            <a href={`/posts/${slug}`}>{title}</a>
+          </li>
+        ))
+      }
+    </ul>
+  </body>
+</html>
+
+```
+
+Fetching via the content API returns an array of objects containing data for each post. The `fields` query parameter tells Builder which data is included (see highlighted code). `slug` and `title` should match the names of the custom data fields you've added to your Builder model.
+
+The `posts` array returned from the fetch displays a list of blog post titles on the home page. The individual page routes will be created in the next step.
+
+:::tip[framework integrations]
+If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use [one of Builder's integrations](https://www.builder.io/m/integrations) as an alternative to making raw fetch calls through the REST API.
+:::
+
+:::tip[API explorer]
```

Comment:
Did you really think you were going to sneak TWO ADJACENT TIPS past me? 😂 

This is visually very distracting, and I remember that I moved this because the link is at the bottom of the page already, and, we don't want to inject random troubleshooting (implying they should be having problems) in the middle of the instructions.

This is another "strongly dislike; bad practice" (on multiple fronts!) that you're trying to get me to accept in my docs.  :smile: